modules/form-completion-status/src/main/java/io/uhndata/cards/formcompletionstatus/ConditionalSectionUtils.java

```diff
@@ -117,6 +117,46 @@
             }
         };
 
+    @SuppressWarnings("serial")
+    private static final Map<Integer, java.util.function.BiFunction<Object, Object, Boolean>> LTE_OPERATORS =
+        new HashMap<Integer, java.util.function.BiFunction<Object, Object, Boolean>>()
+        {
+            {
+                put(PropertyType.LONG, (a, b) -> toLong(a) <= toLong(b));
+                put(PropertyType.DOUBLE, (a, b) -> toDouble(a) <= toDouble(b));
+                put(PropertyType.DECIMAL, (a, b) -> {
+                    BigDecimal decimalA = toDecimal(a);
+                    BigDecimal decimalB = toDecimal(b);
+                    return decimalA != null && decimalB != null && decimalA.compareTo(decimalB) <= 0;
+                });
+                put(PropertyType.DATE, (a, b) -> {
+                    Calendar dateA = toDate(a);
+                    Calendar dateB = toDate(b);
+                    return dateA != null && dateB != null && (dateA.before(dateB) || dateA.equals(dateB));
+                });
+            }
+        };
+
+    @SuppressWarnings("serial")
+    private static final Map<Integer, java.util.function.BiFunction<Object, Object, Boolean>> GTE_OPERATORS =
+        new HashMap<Integer, java.util.function.BiFunction<Object, Object, Boolean>>()
+        {
+            {
+                put(PropertyType.LONG, (a, b) -> toLong(a) >= toLong(b));
+                put(PropertyType.DOUBLE, (a, b) -> toDouble(a) >= toDouble(b));
+                put(PropertyType.DECIMAL, (a, b) -> {
+                    BigDecimal decimalA = toDecimal(a);
+                    BigDecimal decimalB = toDecimal(b);
+                    return decimalA != null && decimalB != null && decimalA.compareTo(decimalB) >= 0;
+                });
+                put(PropertyType.DATE, (a, b) -> {
+                    Calendar dateA = toDate(a);
+                    Calendar dateB = toDate(b);
+                    return dateA != null && dateB != null && (dateA.after(dateB) || dateA.equals(dateB));
```

Comment:
This is equivalent to `dateB.before(dateA)` or `!dateA.before(dateB)`. I'd use this one:

```suggestion
                    return dateA != null && dateB != null && !dateA.before(dateB);
```